server/src/test/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResourceClientIntegrationTest.java

```diff
@@ -0,0 +1,108 @@
+package keywhiz.service.resources.automation.v2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.jackson.Jackson;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.UUID;
+import keywhiz.IntegrationTestRule;
+import keywhiz.KeywhizService;
+import keywhiz.TestClients;
+import keywhiz.api.automation.v2.CreateSecretRequestV2;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static keywhiz.TestClients.clientRequest;
+import static keywhiz.client.KeywhizClient.JSON;
+import static org.junit.Assert.assertEquals;
+
+public class BackfillRowHmacResourceClientIntegrationTest {
+  private static final ObjectMapper mapper =
+      KeywhizService.customizeObjectMapper(Jackson.newObjectMapper());
+  private static final Base64.Encoder encoder = Base64.getEncoder();
+
+  OkHttpClient mutualSslClient;
+
+  @ClassRule public static final RuleChain chain = IntegrationTestRule.rule();
+
+  @Before public void setUp() {
+    mutualSslClient = TestClients.mutualSslClient();
+  }
+
+  @Test public void backfillSecretHmacRequiresAuthentication() throws Exception {
+    String secretName = UUID.randomUUID().toString();
+    assertEquals(401, backfillSecretHmac(TestClients.unauthenticatedClient(), secretName, null).code());
+  }
```

Comment:
This test uses a non-authenticated client and verifies that the response is a 401 Unauthorized. To facilitate this, I added an override which allows us to specify the client to be used rather than always using the authenticated client.